presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/DriverConnectionFactory.java

```diff
@@ -31,58 +30,48 @@
     private final Driver driver;
     private final String connectionUrl;
     private final Properties connectionProperties;
-    private final Optional<String> userCredentialName;
-    private final Optional<String> passwordCredentialName;
+    private final ExtraCredentialProvider extraCredentialProvider;
 
-    public DriverConnectionFactory(Driver driver, BaseJdbcConfig config, CredentialProvider credentialProvider)
+    public DriverConnectionFactory(Driver driver, BaseJdbcConfig config, ExtraCredentialProvider extraCredentialProvider)
```

Comment:
I don't think that `DriverConnectionFactory` cares about who provides credentials. Use `CredentialProvider` here.